cmake/Jucer2CMake/main.cpp

```diff
@@ -0,0 +1,253 @@
+// Copyright (c) 2017 Alain Martin
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#include "JuceHeader.h"
+
+#include "Source/Utility/jucer_PresetIDs.h"
+
+#include <iostream>
+#include <string>
+#include <vector>
+#include <tuple>
+#include <map>
+#include <list>
+#include <fstream>
+
+void printError(const std::string &error) {
+  std::cerr << "error: " << error << std::endl;
+}
+
+int main(int argc, char *argv[]) {
+  // Test initial conditions
+  if (argc != 3) {
+    std::cerr << "usage: Jucer2CMake <jucer_project_file> "
+                 "<reproducer_library_path>"
+              << std::endl;
+    return 1;
+  }
+
+  // Parse command line arguments
+  std::vector<std::string> args{argv, argv + argc};
+  const auto &jucerFilePath = args.at(1);
+  const auto jucerFolderPath = juce::File(jucerFilePath)
+                                   .getParentDirectory()
+                                   .getFullPathName()
+                                   .toStdString();
+  const auto &reproducerLibraryPath = args.at(2);
+  const auto currentWorkingDirectory =
+      juce::File::getCurrentWorkingDirectory().getFullPathName().toStdString();
+  const auto cmakeProjectJucerPrefix =
+      juce::File(jucerFilePath).getFileNameWithoutExtension().toStdString() +
+      "_jucer";
```

Comment:
@McMartin The closest I found is Identifier::isValidIdentifier :

Since Identifiers are used as a script variables and XML attributes, they should only contain alphanumeric characters, underscores, or the '-' and ':' characters.

Is it enough for us ?